### DISCUSSION TITLE
How to build a custom lithops runtime for knative?

### DISCUSSION THREAD

**Discussion by faraz-ahmd** [lithops-cloud/lithops#892] [Q&A] [answered]
Hi,

Can someone please explain what should be the contents of the lithops/ directory and the lithops-knative.zip file when building a custom runtime using the skeleton [Dockerfile](https://github.com/lithops-cloud/lithops/blob/4b9ff7d625199f1b6699ed1b6bf78597279c33cc/runtime/knative/Dockerfile) for knative?

```
# Copy Lithops proxy and lib to the container image.
ENV APP_HOME /lithops
WORKDIR $APP_HOME

COPY lithops_knative.zip .
RUN unzip lithops_knative.zip && rm lithops_knative.zip

CMD exec gunicorn --bind :$PORT --workers $CONCURRENCY --timeout $TIMEOUT lithopsproxy:proxy
```

**Comment by JosepSampe** [comment, marked as answer]
For building a custom runtime you must follow the instructions located in the [runtime/](https://github.com/lithops-cloud/lithops/tree/master/runtime) folder.

The .zip file is automatically created when you excute the `lithops runtime vuild ...` command

**Reply by faraz-ahmd** [reply]
Thanks a lot for the pointer. I was able to build and create runtimes. Most of my issues were due to my setup being behind a proxy, so I had to set environment variables in the dockerfile. 
```
ENV http_proxy "http://web-proxy.net:8080"
ENV https_proxy "http://web-proxy.net:8080"
```

Also, it would be great if the instructions are updated. `lithops runtime deploy` does not work as in the instructions, I used `lithops runtime create` with `lithops, version 2.5.8`.
